# Теория вероятностей и математическая статистика (семинары)
# Урок 10. Дисперсионный анализ

## 1. Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трех группах случайно выбранных спортсменов: Футболисты: 173, 175, 180, 178, 177, 185, 183, 182. Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180. Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170. Данная промежуточная аттестация оценивается по системе "зачет" / "не зачет". "Зачет" ставится, если Слушатель успешно выполнил задание. "Незачет" ставится, если Слушатель не выполнил задание. Критерии оценивания: 1 - Слушатель провел дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов.

In [19]:
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt

y1 = np.array([173, 175, 180, 178, 177, 185, 183, 182])
y2 = np.array([177, 179, 180, 188, 177, 172, 171, 184, 180])
y3 = np.array([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170])

n1 = len(y1)
n2 = len(y2)
n3 = len(y3)

k = 3
n = n1 + n2 + n3

print(f'n1 = {n1} n2 = {n2}  n3 = {n3}')

# Среднее значение
y1_mean = y1.mean()
y2_mean = y2.mean()
y3_mean = y3.mean()


print(f'y1_mean = {y1_mean} y2_mean = {y2_mean} y3_mean = {y3_mean} ')

total = np.concatenate([y1, y2, y3])
print(f'total = {total}')

total_mean = total.mean()
print(f'total_mean = {total_mean}')



n1 = 8 n2 = 9  n3 = 11
y1_mean = 179.125 y2_mean = 178.66666666666666 y3_mean = 172.72727272727272 
total = [173 175 180 178 177 185 183 182 177 179 180 188 177 172 171 184 180 172
 173 169 177 166 180 178 177 172 166 170]
total_mean = 176.46428571428572


In [20]:

S_f = n1 * (y1_mean - total_mean) ** 2 + n2 * (y2_mean - total_mean) ** 2 + n3 * (y3_mean - total_mean) ** 2
S_ost = ((y1 - y1_mean) ** 2).sum() + ((y2 - y2_mean) ** 2).sum() + ((y3 - y3_mean) ** 2).sum()


print(f'S_f = {S_f}   S_ost = {S_ost}')
print(f'S_f + S_ost = {S_f + S_ost}')

S2 = ((total - total_mean) ** 2).sum()
print(f'S2 = {S2}')


S_f = 253.9074675324678   S_ost = 577.0568181818182
S_f + S_ost = 830.964285714286
S2 = 830.9642857142854


In [21]:

D_f = S_f / (k - 1)
D_ost = S_ost / (n - k)

print(f'D_f = {D_f}  D_ost = {D_ost}')

F_n = D_f / D_ost

print(f'F_n = {F_n}')

f = stats.f_oneway(y1, y2, y3)
print(f'{f}')


D_f = 126.9537337662339  D_ost = 23.08227272727273
F_n = 5.500053450812598
F_onewayResult(statistic=5.500053450812596, pvalue=0.010482206918698694)


In [22]:
a = 0.05

F_c = stats.f.ppf(1 - a, (k - 1), (n - k))

print(f'F_c = {F_c}')

F_c = 3.3851899614491687


Давайте проанализируем результат. (F_n = 5.5) > (F_c = 3.385) за счет этих результатов получаем, что отличие среднего роста футболистов, хокеистов и штангистов является статистически значимым.